src/components/MainSidebar/MainSidebar.js

```diff
@@ -19,6 +25,43 @@ const TAB_INPUT = 'INPUT'
 const TAB_PROGRAM = 'PROGRAM'
 const TAB_YOUR = 'YOUR'
 
+const useSelectedDimensions = () => {
```

Comment:
What about exporting this from a separate hook file and letting the components use the hook, rather than passing the `isSelected` function around?

Reply:
I agree that passing the isSelected function around isn't ideal. On the other hand, if we let individual components use this hook, it means that several "instances" of this hook will be doing the exact same computation, which also isn't great. I briefly considered to implement this using a context-provider + hook. This is probably a good solution, because it would allow us to do the computations in one place, and allow all children to import the hook and thus avoid passing things around. I originally decided against it because it would make the code slightly harder to follow. But, upon reflection, it seems that passing props through several layers of components doesn't make for the most readable code ever. So I'll try a solution with a context provider.

Reply:
See implementation in [19bc100](https://github.com/dhis2/event-reports-app/pull/916/commits/19bc100b45b16f6cb89c21325adbebdb4f5a6795)

Reply:
I like the implementation.